### PR TITLE
(feat) finish calender/search by date functionality for meallist

### PR DESCRIPTION
**Pull request by rs214**

### CODE REVIEW

File: client/src/containers/MealList.js [hunk 1]
```diff
@@ -25,6 +28,16 @@ class MealList extends Component {
     this.props.getNearbyMeals(this.props.user.zipcode);
   }
 
+  onInputChange(event) {
+    this.setState({ term: event.target.value })
+  }
+
+  onFormSubmit(event) {
+    console.log(this.props.fetchMealsByDate(this.state.term))
```
Comment (gzeegz) [inline client/src/containers/MealList.js]: `console.log` needs to be removed.

File: client/src/containers/SearchBar.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { fetchMealsByDate } from '../actions';
+import { bindActionCreators } from 'redux';
+import RaisedButton from 'material-ui/RaisedButton';
+// import { Field, reduxForm } from 'redux-form';
+// import {renderDateField} from '../utils/FormHelper';
```
Comment (gzeegz) [inline client/src/containers/SearchBar.js]: Commented code should be removed.

File: client/src/containers/SearchBar.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    this.setState({
+      controlledDate: date
+    });
+  };
+
+  onFormSubmit(event) {
+    var dateobj = new Date(this.state.controlledDate)
+    var date = dateobj.toISOString().substring(0,10)
+    this.props.fetchMealsByDate(date);
+    this.setState({ controlledDate: null })
```
Comment (gzeegz) [inline client/src/containers/SearchBar.js]: Missing some semicolons here.

File: client/src/containers/SearchBar.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    return (
+      <div>
+        <DatePicker
+          hintText="Find a meal by date"
+          autoOk= {true}
+          value={this.state.controlledDate}
+          onChange={this.handleChange}
+        />
+        <span>
+        <RaisedButton onClick={this.onFormSubmit}>Submit</RaisedButton>
```
Comment (gzeegz) [inline client/src/containers/SearchBar.js]: Indent the button component.

File: client/src/reducers/mealsReducer.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { FETCH_MEALS, FETCH_MEALDETAIL, CREATE_MEAL, REVIEW_MEAL, FETCH_MEALS_BY_CHEF, GET_NEAR_BY_MEALS } from '../actions/index';
+import { FETCH_MEALS, FETCH_MEALDETAIL, CREATE_MEAL, REVIEW_MEAL, FETCH_MEALS_BY_CHEF, GET_NEAR_BY_MEALS, FETCH_MEALS_BY_DATE } from '../actions/index';
```
Comment (gzeegz) [inline client/src/reducers/mealsReducer.js]: This line is getting really long. Let's break it over multiple lines.

**Review by gzeegz — CHANGES_REQUESTED** [review]

**Review by gzeegz — APPROVED** [review]
LGTM

**Comment by obay-mardini** [comment]
Please could you resolve those conflicts

**Review by obay-mardini — CHANGES_REQUESTED** [review]
could resolve the merge conflicts in this PR

**Comment by rs214** [comment]
yes let me rebase, fix

**Pull request closed without merging**